.github/workflows/release.yml

```
@@ -5,30 +5,47 @@ on:
     types:
       - created
 
+env:
+  AWS_S3_BUCKET: ksp-ckan
+
 jobs:
-  release:
-    runs-on: ubuntu-latest
+  test-release:
+    uses: ./.github/workflows/test.yml
 
-    container:
-      image: mono:latest
+  smoke-inflator:
+    uses: ./.github/workflows/smoke.yml
 
+  sign-release:
+    runs-on: ubuntu-latest
+    needs:
+      - test-release
+      - smoke-inflator
+    outputs:
+      artifact-url: ${{steps.sign.outputs.signing-request-id }}
     steps:
-      - name: Installing checkout/build dependencies
-        run: apt-get update && apt-get install -y git make sed libplist-utils xorriso gzip fakeroot lintian rpm wget jq dpkg-dev gpg createrepo
-      - uses: actions/checkout@v3
-
-      - name: Setup .NET Core
-        uses: actions/setup-dotnet@v3
+      - uses: signpath/github-action-submit-signing-request@v0.3
+        id: sign
         with:
-          dotnet-version: '7'
-      - name: Installing runtime dependencies
-        run: apt-get install -y xvfb
-
-      - name: Build ckan.exe and netkan.exe
-        run: ./build --configuration=Release
-      - name: Run tests
-        run: xvfb-run ./build test+only --configuration=Release --where="Category!=FlakyNetwork"
+          api-token: ${{ secrets.SIGNPATH_API_TOKEN }}
+          organization-id: 0cd9fc3f-b78d-4214-b152-b2e93c952e14
+          project-slug: CKAN
+          signing-policy-slug: test-signing
+          github-artifact-name: Release-repack-unsigned
+          artifact-configuration-slug: release
+          wait-for-completion: true
 
+  upload-nuget:
+    runs-on: ubuntu-latest
+    needs:
+      - test-release
+      - smoke-inflator
+    outputs:
+      artifact-url: ${{steps.sign.outputs.signing-request-id }}
+    steps:
+      - uses: actions/upload-artifact@v4
```

```suggestion
      - uses: actions/download-artifact@v4
```

Is this using the right action? I would expect downloading here, not uploading.